src/components/reusable/table/story-helpers/table.settings.ts

```diff
@@ -0,0 +1,364 @@
+    },
+    {
+      id: 'col13',
+      order: 13,
+      colName: 'COLUMN 13',
+      locked: false,
+      visible: true,
+      align: 'center',
+    },
+    {
+      id: 'col14',
+      order: 14,
+      colName: 'COLUMN 14',
+      locked: false,
+      visible: true,
+      align: 'center',
+    },
+    {
+      id: 'col15',
+      order: 15,
+      colName: 'COLUMN 15',
+      locked: false,
+      visible: true,
+      align: 'center',
+    },
+    {
+      id: 'col16',
+      order: 16,
+      colName: 'COLUMN 16',
+      locked: false,
+      visible: false,
+      align: 'center',
+    },
+    {
+      id: 'col17',
+      order: 17,
+      colName: 'COLUMN 17',
+      locked: false,
+      visible: false,
+      align: 'center',
+    },
+    {
+      id: 'col18',
+      order: 18,
+      colName: 'COLUMN 18',
+      locked: false,
+      visible: false,
+      align: 'center',
+    },
+    {
+      id: 'col19',
+      order: 19,
+      colName: 'COLUMN 19',
+      locked: false,
+      visible: false,
+      align: 'center',
+    },
+    {
+      id: 'col20',
+      order: 20,
+      colName: 'COLUMN 20',
+      locked: false,
+      visible: false,
+      align: 'center',
+    },
+  ];
+
+  override firstUpdated() {
+    this.columnSetting = this.shadowRoot?.querySelector('story-column-setting');
+  }
+
+  handleSettingsClick(event: any) {
+    event.preventDefault();
+    event.stopPropagation();
+
+    this.open = !this.open;
+    this.requestUpdate();
+  }
+
+  handleClose(e: CustomEvent) {
+    e.preventDefault();
+    e.stopPropagation();
+
+    if (e.detail.returnValue === 'ok') {
+      const updatedColumns = this.columnSetting.getRows();
+
+      // Update the columns
+      updatedColumns.forEach((updatedColumn: any) => {
+        const originalColumn = this.columns.find(
+          (column: any) => column.id === updatedColumn.id
+        );
+        if (originalColumn) {
+          originalColumn.visible = updatedColumn.visible;
+          originalColumn.locked = updatedColumn.locked;
+        }
+      });
+      this.requestUpdate();
+      return;
+    }
+  }
+
+  _handleSearch(e: any) {
+    confirm('Search for ' + e.target.value);
+  }
+
+  override render() {
+    // sort the columns by order
+    this.columns.sort((a: any, b: any) => a.order - b.order);
+
+    // Find the locked column
+    const lockedCol = this.columns.find((col: any) => col.locked);
+
+    if (lockedCol) {
+      // Remove the locked col from its current position
+      const index = this.columns.indexOf(lockedCol);
+      this.columns.splice(index, 1);
+
+      // Add the locked col to the top
+      this.columns.unshift(lockedCol);
+    }
+
+    return html`
+      <style>
+        .sr-only {
+          position: absolute;
+          width: 1px;
+          height: 1px;
+          padding: 0;
+          margin: -1px;
+          overflow: hidden;
+          clip: rect(0, 0, 0, 0);
+          border: 0;
+        }
+      </style>
```

Comment:
its better to add styles in separate .scss file. Just suggestion

Reply:
Well this is just a story, not a real component. Are we trying to formalizing story components? I'm fine if we all agree on this.
But just FYI - Today I saw inline styles in Card stories too.

Reply:
`sr-only` can be imported from Foundation/Visibility, but I think a bigger issue is the Settings button in the global filter is not used as the anchor for the Side Drawer, that would make more sense IMO. I noticed I also have to click "Settings" twice to re-open the drawer as well.